src/fortran/fdb.f90

```diff
@@ -1,39 +1,329 @@
+
+   interface
+      integer(kind=c_int) function fdb_new_listiterator(it) bind(C,name='fdb_new_listiterator')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr
+         type(c_ptr), intent(in)      :: it
+      end function fdb_new_listiterator
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_listiterator_next(it) bind(C,name='fdb_listiterator_next')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr, c_char, c_bool
+         type(c_ptr), intent(in), value               :: it
+      end function fdb_listiterator_next
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_delete_listiterator(it) bind(C,name='fdb_delete_listiterator')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr
+         type(c_ptr), intent(in), value      :: it
+      end function fdb_delete_listiterator
+   end interface
+
+   interface
+      integer(kind=c_int) function fdb_listiterator_attrs(it, uri, off, len) bind(C,name='fdb_listiterator_attrs')
+         use, intrinsic :: iso_c_binding, only : c_int, c_ptr, c_int, c_char
+         type(c_ptr), intent(in), value      :: it
+         character(kind=c_char, len=1), dimension(*), INTENT(inout) :: uri
+         integer(kind=c_int), intent(inout) :: off
+         integer(kind=c_int), intent(inout) :: len
+      end function fdb_listiterator_attrs
+   end interface
+
+   CONTAINS
+   SUBROUTINE get_value_of_key(igrib, keyname, keyvalue_str, keyvalue_int)
+      character(len=*), INTENT(IN)         :: keyname
+      character(len=128), INTENT(OUT)    :: keyvalue_str
+      integer, INTENT(INOUT), OPTIONAL     :: keyvalue_int
+      integer, INTENT(IN)                 :: igrib
+      integer                             :: res
+      integer                             :: is_missing
+      call codes_is_missing(igrib, trim(keyname), is_missing);
+      if (is_missing /= 1) then
+         ! key value is not missing so get values
+         if (present(keyvalue_int)) then
+            call codes_get_int(igrib, keyname, keyvalue_int)
+            write (keyvalue_str, "(I4)") keyvalue_int
+            keyvalue_str=trim(adjustl(keyvalue_str))
+         else
+            call codes_get(igrib, keyname, keyvalue_str)
+         end if
+         write (*, *) keyname, '=', keyvalue_str
+      else
+         write (*, *) keyname, ' is missing from data.'
+      end if
+   END SUBROUTINE
+
+   SUBROUTINE copy_s2a(a,s)   ! copy s(1:Clen(s)) to char array
+      CHARACTER(*),INTENT(IN) :: s
+      CHARACTER :: a(LEN(s)+1)
+      INTEGER :: i
+      DO i = 1,LEN(s)
+         a(i) = s(i:i)
+      END DO
+      a(LEN(s)+1) = char(0)
+   END SUBROUTINE copy_s2a
+
+   SUBROUTINE copy_a2s(a,s)  
+      CHARACTER,INTENT(IN) :: a(:)
+      CHARACTER(len=*),INTENT(INOUT)  :: s
+      INTEGER :: i
+      DO i = 1,SIZE(a)
+         s(i:i) = 'z' ! a(i) ! FAILS HERE
+      END DO
+   END SUBROUTINE copy_a2s
+
+   FUNCTION index_chararray(a,s) result(idx)  
```

Comment:
uhmm this function is not used anywhere, isnt? Can we delete it?

Reply:
deleted